criu/cr-restore.c

```diff
@@ -3201,6 +3201,43 @@ static void rst_reloc_creds(struct thread_restore_args *thread_args,
 	thread_args->creds_args = args;
 }
 
+static bool groups_match(gid_t* groups, int n_groups)
+{
+	int n, len;
+	bool ret;
+	gid_t* gids;
+
+	n = getgroups(0, NULL);
+	if (n == -1) {
+		pr_perror("Failed to get number of supplementary groups");
+		ret = false;
+	}
+	if (n != n_groups)
+		return false;
+	if (n == 0)
+		return true;
+
+	len = n * sizeof(gid_t);
+	gids = xmalloc(len);
+	if (gids == NULL)
+		return false;
+
+	n = getgroups(n, gids);
+	if (n == -1) {
+		pr_perror("Failed to get supplementary groups");
+		ret = false;
+	} else {
+		/*
+		* Because getgroups sorts gids,
+		* memcmp compares two sorted arrays
+		*/
```

Comment:
The comment is short enough to fit into a single line ;-)
I'll update it myself during the pull.

And, for future reference, in multi-line comments the `*` should be aligned:

```
/*
 * This comment is very long and 
 * spans multiple lines
 */
```

Reply:
Ah, thank you for letting me know + for manually merging! I'll be sure my future comments a**_line_** with these guidelines :^)